File: StartAppFrontend/app/src/main/java/com/skyletto/startappfrontend/domain/entities/Tag.java
package com.skyletto.startappfrontend.domain.entities;

import java.util.Objects;

import androidx.room.Entity;
import androidx.room.PrimaryKey;

@Entity(tableName = "tags")
public class Tag {
    @PrimaryKey
    private long id;
    private String name;

    public long getId() {
        return id;
    }

    public void setId(long id) {
        this.id = id;
    }

    public String getName() {
        return name;
    }

    public void setName(String name) {
        this.name = name;
    }

    @Override
    public boolean equals(Object o) {
        if (this == o) return true;
        if (o == null || getClass() != o.getClass()) return false;
        Tag tag = (Tag) o;
        return id == tag.id &&
                name.equals(tag.name);
    }

    @Override
    public int hashCode() {
        return Objects.hash(id, name);
    }
}

File: StartAppFrontend/app/src/main/java/com/skyletto/startappfrontend/domain/entities/Tag.kt
package com.skyletto.startappfrontend.domain.entities;

import androidx.room.ColumnInfo
import androidx.room.Entity;
import androidx.room.PrimaryKey;
import com.squareup.moshi.Json

@Entity(tableName = "tags")
data class Tag(
        @PrimaryKey
        @ColumnInfo(name = "tId")
        val id: Long,
        val name: String
)
